File: backend_samples/library/java/AzureAIVisionFaceDeviceAttestation/src/main/java/com/azure/ai/vision/face/deviceattestation/crypto/Base64Utils.java
package com.azure.ai.vision.face.deviceattestation.crypto;

import java.util.Base64;

/** URL-safe / padded base64 decoding used across the wire protocol. */
public final class Base64Utils {

    /** Decode standard or URL-safe base64, tolerating missing padding. */
    public static byte[] decode(String value) {
        String normalized = value.replace('-', '+').replace('_', '/');
        int remainder = normalized.length() % 4;
        if (remainder != 0) {
            normalized = normalized + "====".substring(remainder);
        }
        return Base64.getDecoder().decode(normalized);
    }

    private Base64Utils() {
    }
}

File: backend_samples/library/dotnet/AzureAIVisionFaceDeviceAttestation/src/Crypto/Base64Utils.cs
namespace Azure.AI.Vision.Face.DeviceAttestation.Crypto;

internal static class Base64Utils
{
    public static byte[] Decode(string value)
    {
        var normalized = value.Replace('-', '+').Replace('_', '/');
        normalized = normalized.PadRight(normalized.Length + ((4 - normalized.Length % 4) % 4), '=');
        return Convert.FromBase64String(normalized);
    }
}
